### ISSUE TITLE
Typings

### ISSUE THREAD

**Issue by oleg-koval** [customerio/customerio-node#54]
Hi!
How to use typings of `customerio-node`?

thanks!

**Comment by mike-engel** [comment]
Hi @oleg-koval! Could you provide more detail on the trouble you're having? Examples would help a lot. Thanks!

**Comment by nickzelei** [comment]
@mike-engel , looking at the repo, I don't think the project is configured quite right for typescript.
I don't see any index file in the repo at all, as well, the `package.json` has no entry point or typings entrypoint defined.

It looks like your build step is building the typescript files to `dist`.

You should update your package.json to include this:
```
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
```
Or whichever you define is the entrypoint.

I think previously you had the `track.js` as the entrypoint. It would be great if `api.ts` could also be exported.

A potential `index.ts`
```
export * from './track';
export * from './api'; // however this might get confused as there is also a folder named api
```

I also noticed the old javascript files are still present in the repo. Those shouldn't be necessary anymore given the typescript files :)

**Comment by mike-engel** [comment]
Thanks for the breakdown @nickzelei! Whatever tests I ran prior to releasing 2.1.0 obviously weren't enough. Apologies for not testing the typings more thoroughly!

The current structure of this package stems from earlier api versions—and while I agree that it's not the best that only `track` is the default entry point, I also don't want to break the API for existing consumers just to add typescript definitions. That's why I went the route I did for #49.

The three "older" javascript files are there on purpose to provide the current API for all clients. This allows for the following imports which we use throughout the README and the examples dir:

```js
// Track API
import TrackApi from 'customerio-node';

// App API
import AppAPI from 'customerio-node/api';

// Region configs
import { RegionUS, RegionEU } from 'customerio-node/regions';
```

Again, I know that this is not ideal, but at this point we're trying not to make any breaking changes for existing clients unless we have a really good reason to.

That being said, I've just published `customerio-node@2.1.1-beta.1` (#56) which I believe keeps the same API but exposes the proper typings for consumers writing code in Typescript. If y'all could take a look and check it with your codebases, that would be wonderful. You can also install it with `customerio-node@beta`.

In my own testing, `tsc` compiles a tiny example without any errors with that exact version. I'm hoping that it will for you as well.

Again, thank you for your patience, and apologies for the false start with typescript typings!

**Comment by nickzelei** [comment]
Thanks for the detailed response @mike-engel, I totally get not wanting to break the API for consumers :smile: 

I think that your tests are fine, the reason they didn't catch the entrypoint issue is because they import specifically from the files themselves since there is no single point of entry for this module. Which makes sense given the context you just gave us.

**Comment by mike-engel** [comment]
Typings should be fixed with `2.1.1` via #56!
